### PR TITLE
Multihit WS & OaT jump audit

### PR DESCRIPTION
**Pull request by WinterSolstice8**
**_I affirm:_**
- [x] I understand that if I do not agree to the following points by completing the checkboxes my PR will be ignored.
- [x] I understand I should leave resolving conversations to the LandSandBoat team so that reviewers won't miss what was said.
- [x] I have read and understood the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md) and the [Code of Conduct](https://github.com/LandSandBoat/server/blob/base/CODE_OF_CONDUCT.md).
- [x] I have _**tested my code and the things my code has changed**_ since the last commit in the PR and will test after any later commits.

## What does this pull request do?

Audits usage of multihit WS such as follows:

Mainhand, offhand, mainhand extra WS hits (>1 hit WS), mainhand DA/TA/QA/etc procs, offhand DA/TA/QA/etc procs
All WS can proc DA/TA/QA etc twice, but the second proc requires >= 2 base hits on WS or an offhand
Offhand hits now use offhand hit damage on DA/TA/QA etc
WSD no longer applies to Jump.
Skillups on offhand hand now use the appropriate skill if different.

## Steps to test these changes

Use weaponskills, see stuff look pretty normal.
Use Jump and see DA/TA/QA and OaX work properly with normal TP returns.

### CODE REVIEW

File: scripts/globals/weaponskills.lua [hunk 1]
```diff
@@ -216,51 +215,35 @@ local function getMultiAttacks(attacker, target, wsParams)
...
-    -- whereas dual wiedling gets 2 "initial" hits, but `numHits` for e.g. dancing edge is respected as though only one "initial" was performed
-    -- tl;dr: DW gives an extra WS hit everytime, H2H simply ensures every WS gets at least 2 hits (yes backhand blow and dragon kick are both 2-hit ws...)
     numHits = numHits + bonusHits
-    if attacker:getOffhandDmg() > 0 then
-        numHits = numHits + 1
-    end
 
-    -- cap total hits at 8
-    return utils.clamp(numHits, 1, 8)
+    return numHits
```
Comment (Xaver-DaRed) [inline scripts/globals/weaponskills.lua:246]: No clamp needed?

Comment (WinterSolstice8) [reply scripts/globals/weaponskills.lua:246]: Clamp is moved to the hit count loops, it's not possible for that function to return > 4 anyway

**Review by Xaver-DaRed — DISMISSED** [review]

**Comment by WinterSolstice8** [comment]
I just realized that there's a potential bug with TP gain here, drafted

**Comment by WinterSolstice8** [comment]
Please re-review, should be good to go.

**Review by Xaver-DaRed — APPROVED** [review]

**Pull request merged**